pkg/apis/networking/v1alpha1/clusteringress_types.go

```diff
@@ -0,0 +1,375 @@
+	ServiceNamespace string `json:"serviceNamespace"`
+
+	// Specifies the name of the referenced service.
+	ServiceName string `json:"serviceName"`
+
+	// Specifies the port of the referenced service.
+	ServicePort intstr.IntOrString `json:"servicePort"`
+}
+
+// ClusterIngressBackend describes all endpoints for a given service and port.
+type ClusterIngressBackendSplit struct {
+	// Specifies the backend receiving the traffic split.
+	Backend *ClusterIngressBackend `json:"backend"`
+
+	// Specifies the split percentage, a number between 0 and 100.  If
+	// only one split is specified, we default to 100.
+	//
+	// NOTE: This differs from K8s Ingress to allow percentage split.
+	Percent int `json:"percent,omitempty"`
+}
+
+var _ apis.Validatable = (*ClusterIngress)(nil)
+var _ apis.Defaultable = (*ClusterIngress)(nil)
+
+// Check that ClusterIngress implements the Conditions duck type.
+var _ = duck.VerifyType(&ClusterIngress{}, &duckv1alpha1.Conditions{})
+
+// Check that ClusterIngress implements the Generation duck type.
+var emptyGen duckv1alpha1.Generation
+var _ = duck.VerifyType(&ClusterIngress{}, &emptyGen)
+
+func (ci *ClusterIngress) GetGeneration() int64 {
+	return ci.Spec.Generation
+}
+
+func (ci *ClusterIngress) SetGeneration(generation int64) {
+	ci.Spec.Generation = generation
+}
+
+func (ci *ClusterIngress) GetSpecJSON() ([]byte, error) {
+	return json.Marshal(ci.Spec)
+}
+
+func (ci *ClusterIngress) GetGroupVersionKind() schema.GroupVersionKind {
+	return SchemeGroupVersion.WithKind("ClusterIngress")
+}
+
+// GetConditions returns the Conditions array. This enables generic handling of
+// conditions by implementing the duckv1alpha1.Conditions interface.
+func (cis *ClusterIngressStatus) GetConditions() duckv1alpha1.Conditions {
+	return cis.Conditions
+}
+
+// SetConditions sets the Conditions array. This enables generic handling of
+// conditions by implementing the duckv1alpha1.Conditions interface.
+func (cis *ClusterIngressStatus) SetConditions(conditions duckv1alpha1.Conditions) {
+	cis.Conditions = conditions
+}
+
+func (cis *ClusterIngressStatus) GetCondition(t duckv1alpha1.ConditionType) *duckv1alpha1.Condition {
+	return clusterIngressCondSet.Manage(cis).GetCondition(t)
+}
+
+func (cis *ClusterIngressStatus) InitializeConditions() {
+	clusterIngressCondSet.Manage(cis).InitializeConditions()
+}
+
+func (cis *ClusterIngressStatus) MarkNetworkConfigured() {
+	clusterIngressCondSet.Manage(cis).MarkTrue(ClusterIngressConditionNetworkConfigured)
+}
+
+// MarkLoadBalancerReady marks the Ingress with ClusterIngressConditionLoadBalancerReady,
+// and also populate the address of the load balancer.  We allow multiple load balancers
+// address, but currently our Istio implementation only needs/has one so this signature
+// is better taking []LoadBalancerIngress, as we can make sure exactly one address is
+// provided.
+func (cis *ClusterIngressStatus) MarkLoadBalancerReady(lb LoadBalancerIngress) {
```

Comment:
I am not sure if we should mention Istio here since it is related to the implementation.
And I prefer using []LoadBalancerIngress since it is more general.

Reply:
Fixed.